### ISSUE TITLE
[BUG] `fitBounds` doesn't cause new tiles to load

### ISSUE THREAD

**Issue by billyz313** [fleaflet/flutter_map#1563]
### What is the bug?

When I use the .fitBounds method the map moves to the correct location, however the tiles which are supposed to be displayed on the map do not load.  If I slightly pan or zoom the map the tiles show up.  This happens in the web version when running thru the Chrome browser.  It works fine in the android version.  You can see it happening in the image below.

![fitBoundsExample](https://github.com/fleaflet/flutter_map/assets/1868821/d94f052e-937b-4c24-8ce6-f6f5950c1ea9)

### How can we reproduce it?

I removed the small bit of code from the search and hard coded the values so the code to reproduce can be as simple as 

```
widget.controller.fitBounds(LatLngBounds(
                      LatLng(46.3471572, 0.0527585),
                      LatLng(48.9410263, 3.1286392)));
```

I work around it by programmatically moving the map directly after the fitBounds and it works, but I feel like it should be able to just fitBounds to show the tiles.

### Platforms

Windows, visual studio code, Chrome(web-javascript), debug

### Severity

Obtrusive: Prevents normal functioning but causes no errors in the console

**Comment by JaffaKetchup** [comment]
Hi @billyz313,

Thanks for the report. This is strange that it only affects certain platforms.
Can you please confirm your flutter_map version?

If you have the spare time, please try depending on #1551, as that may or may not solve this issue. A pubspec.yaml snippet to do this is:

```yaml
dependency_overrides:
    flutter_map:
        git:
            url: https://github.com/rorystephenson/flutter_map.git
            ref: flutter-map-state-refactor
```

**Comment by billyz313** [comment]
My pleasure, I also found it strange.  

In my pubspec i currently have flutter_map: ^4.0.0.

Unfortunately I can't use the flutter_map in the repo because I am using flutter_map_dragmarker and flutter_map_line_editor which throw a bunch of errors all seeming to lead back to stuff like "The getter 'FlutterMapState' isn't defined for the class 'DragMarkers'."  So I assume those just are not ready for the new version...

**Comment by billyz313** [comment]
@ibrierley Thank you, I do have the updated ones for v5, however they seem to not work with the dependency_overrides suggestion from above

```
dependency_overrides:
    flutter_map:
        git:
            url: https://github.com/rorystephenson/flutter_map.git
            ref: flutter-map-state-refactor
```

**Comment by ozkayas** [comment]
I am facing the same issue on flutter web. Tiles are loaded at first, then after getting positions and fitBounds is called, tiles disappear.
I am using flutter_map: 5.0.0
Not sure what triggered this issue.


https://github.com/fleaflet/flutter_map/assets/63163416/937f5c63-c27f-4621-b6a3-4b188cd5a68a

**Comment by billyz313** [comment]
@ozkayas I have worked around the issue by adding a move after the fitBounds.  The move actually does not move at all, I pass it the same lat, lon, and zoom, but it refreshes the tiles.  This of course is not the fix for the bug, but it is a workaround until a fix is introduced.   One caveat I guess is that I have both the bounding box and the center lat, lon.  I reckon using widget.controller.center.latitude and widget.controller.center.longitude will also give the same results. 
Here is my code:
```
widget.controller.fitBounds(bounds);
widget.controller.move(
    LatLng(double.parse(result["lat"]),
        double.parse(result['lon'])),
    widget.controller.zoom);
```

**Comment by shahpasandar** [comment]
I have the same issue on Android/iOS versions of my app with `flutter_map: ^5.0.0`.

```
Flutter 3.10.5 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 796c8ef792 (7 weeks ago) • 2023-06-13 15:51:02 -0700
Engine • revision 45f6e00911
Tools • Dart 3.0.5 • DevTools 2.23.1
```

**Comment by JaffaKetchup** [comment]
Although I have not yet gotten around to reproducing this, with the extra reports in this comments thread and the report at #1601, I'm marking this as P1 Important.

It appears as though this issue affects more platforms than just web, so it's unclear why OP could only reproduce on web.

**Comment by sikandernoori** [comment]
### How can we reproduce this on IOS?

Run example app go to Animated MapController page, zoom in a bit then click `Fit Bounds` button on screen.


https://github.com/fleaflet/flutter_map/assets/16399272/40704881-3c3b-4b7b-bea1-273b5863c613

**Comment by slawekkrol** [comment]
Temporarily solution for that (version v5.0.0) until fix:

    _mapController
      ..fitBounds(
          LatLngBounds(
            const LatLng(46.3471572, 0.0527585),
            const LatLng(48.9410263, 3.1286392),
          ),
          options: FitBoundsOptions(
            padding: EdgeInsets.only(top: 40),
          ))
      ..move(_mapController.center, _mapController.zoom + 0.00001);

**Comment by afrish** [comment]
This is a critical defect that totally prevents us from using fitBounds() method on iOS and Android

**Comment by JaffaKetchup** [comment]
@afrish Please see the workaround above for now. Call `move` with a tiny change in the zoom level - it won't be noticable, but it will cause tiles to load.

**Comment by afrish** [comment]
> @afrish Please see the workaround above for now. Call `move` with a tiny change in the zoom level - it won't be noticable, but it will cause tiles to load.

Ok, thank you, I will try this. I would be glad to help fixing this, but I have too little knowledge in the library internals...

**Comment by JaffaKetchup** [comment]
This is fixed in #1626.

I'm not sure exactly why the event source being `fitCamera` does not cause tiles to refresh, but `mapController` does. Note that in v6, this method is deprecated in favour of `fitCamera`, which is more flexible.

**Comment by stevenv-synap** [comment]
@JaffaKetchup Thanks so much for your hard work. Just came to have a look around for some guidance, and ended up seeing a solution has already been submitted for merging. Much appreciated! <3
